js/packages/mobile-wallet-adapter-protocol-web3js/src/transact.ts

```diff
@@ -158,3 +178,18 @@ export async function transact<TReturn>(
     };
     return await baseTransact(augmentedCallback, config);
 }
+
+function validateSecureEndpointSpecificURI(walletUriBase?: string) {
+    if (walletUriBase) {
+        let url: URL | null = null;
+        try {
+            url = new URL(walletUriBase);
+        } catch {} // eslint-disable-line no-empty
```

Comment:
If we're saying that we can't let through non-https URLs we probably also shouldn't let through URLs that straight up don't parse.

Reply:
Done